beastify/popup/choose_beast.js

```diff
@@ -1,54 +1,81 @@
-/*
-Given the name of a beast, get the URL to the corresponding image.
-*/
-function beastNameToURL(beastName) {
-  switch (beastName) {
-    case "Frog":
-      return browser.extension.getURL("beasts/frog.jpg");
-    case "Snake":
-      return browser.extension.getURL("beasts/snake.jpg");
-    case "Turtle":
-      return browser.extension.getURL("beasts/turtle.jpg");
-  }
-}
+/**
+ * CSS to hide everything on the page,
+ * except for elements that have the "beastify-image" class.
+ */
+const hidePage = `body > :not(.beastify-image) {
+                    display: none;
+                  }`;
 
-/*
-Listen for clicks in the popup.
+/**
+ * When the popup loads, inject a content script into the active tab,
+ * and add a clck handler.
+ */
+browser.tabs.executeScript({file: "/content_scripts/beastify.js"})
+.then(() => {
```

Comment:
1. This approach won't work if the tab unloads/reloads while the popup has been opened.
2. And if the extension is unable to access the document (e.g. privileged URLs such as `about:preferences`), then the buttons won't do anything upon "click".
3. Lastly, if the page is still loading, then the content script won't be injected (until the document has finished, because `runAt` is not set, so it defaults to `document_idle`) and the button will also appear to not handle clicks.

The second issue should be fixed, e.g. by adding a second callback to `.then` and then updating the UI as appropriate.
To keep the example simple you can just document the other limitations (the situations are quite rare anyway).